pkg/codegen/internal/test/testdata/resource-property-overlap/schema.json

```diff
@@ -5,7 +5,7 @@
     "example::Rec": {
       "properties": {
         "rec": {
-          "$ref": "#/types/example::Rec"
+          "$ref": "#/resources/example::Rec"
```

Comment:
🤔 it sounds like this used to be an invalid schema?

Reply:
I _think_ that this was originally an unexpected schema. We've always allowed for references to undefined types in order to support overlays. With @lblackstone 's recent changes to allow overlays to be represented in a first class fashion in the schema, we might be able to change that.